app/code/Magento/CustomerGraphQl/Model/Customer/UpdateCustomerAccount.php

```diff
@@ -92,32 +91,31 @@ public function __construct(
      */
     public function execute(CustomerInterface $customer, array $data, StoreInterface $store): void
     {
-        if (isset($data['email']) && $customer->getEmail() !== $data['email']) {
-            if (!isset($data['password']) || empty($data['password'])) {
+        $email = $data['email'];
+        $password = $data['password'];
+        $customerId = (int)$customer->getId();
+        if (isset($email) && $customer->getEmail() !== $email) {
+            if (empty($password)) {
                 throw new GraphQlInputException(__('Provide the current "password" to change "email".'));
             }
-
-            $this->checkCustomerPassword->execute($data['password'], (int)$customer->getId());
-            $customer->setEmail($data['email']);
+            $this->checkCustomerPassword->execute($password, $customerId);
+            $customer->setEmail($email);
         }
         $this->validateCustomerData->execute($data);
         $filteredData = array_diff_key($data, array_flip($this->restrictedKeys));
         $this->dataObjectHelper->populateWithArray($customer, $filteredData, CustomerInterface::class);
-
+        
+        $storeId = $store->getId();
         try {
-            $customer->setStoreId($store->getId());
+            $customer->setStoreId($storeId);
```

Comment:
`$customer->setStoreId($store->getId());`
don't need to create a var for this because this don't to be reused.

Reply:
Thank you for your suggestion @Muneebkh2